### PR TITLE
Allow Google Analytics to use gtag.js

### PR DESCRIPTION
**Pull request by ramiel**
The code now uses the new gtag script to track the website on google analytics.
The migration is explained [here](https://developers.google.com/analytics/devguides/collection/gtagjs/migration).
This will help to solve issues such as #375.
As indicated at https://support.google.com/analytics/answer/1008080
the analytics code must be placed in the head tag, so it is moved there.

## Motivation

I need to link my website to google search console so, I need the google analytics code to be setup properly

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

No :)

## Test Plan

The new code is on the <head> now, instead of the body. There were no test for this before and I added none

## Related PRs

None

### CODE REVIEW

**Comment by JoelMarcey** [comment]
> Have you read the Contributing Guidelines on pull requests? No :)

😢 

But, you provided enough detail that we can look at this PR anyway 😄 

I am trying to think if this will cause any backward compatibility problems - don't think so. I added @yangshun to see if he thinks there are any issues here.

**Comment by yangshun** [comment]
Moving it into `<head>` is ok since the `<script>` now has the `async` attribute. The script will start downloading earlier, which is great.

 @ramiel the tests are failing, I think it's because the code has to be prettified first. Run `yarn prettier`.

Note that making this change will break users who are relying on `analytics.js` and the `ga()` method for pushing events as outlined in Google Analytics' [Migration Docs](https://developers.google.com/analytics/devguides/collection/gtagjs/migration). IMO if we want to upgrade this, it is a breaking change and will require a major version bump. @JoelMarcey

**Comment by ramiel** [comment]
So, I rebased and run prettier, now tests pass. For the moment I'd leave this PR as is until you decide what's better. If you say to have an option in siteConfig, I'll create it.
Another alternative could be to only move the original script into head, without migrating to gtag and to leave that change for later. Let me know

**Comment by JoelMarcey** [comment]
@ramiel @yangshun I am ok with adding a siteConfig option for this. Let's go that route. We can call it `gaGtag` (consistent with our current `gaTrackingId`) and have it be boolean `true` for it to take hold.  If you have a better name for the config option, that's good too. And we can put both options in `Head.js`.

Let's do it!

**Comment by ramiel** [comment]
Ok, I'll do these changes soon

File: lib/core/Head.js [hunk 1]
```diff
@@ -95,6 +95,44 @@ class Head extends React.Component {
...
+            <script
+              async
+              src={`https://www.googletagmanager.com/gtag/js?id=${
+                this.props.config.gaTrackingId
+              }`}
+            />
+          )}
+        {this.props.config.gaTrackingId &&
+          this.props.config.gaGtag && (
+            <script
```
Comment (JoelMarcey) [inline lib/core/Head.js]: Couldn't use combine the the two `<script>` into one `gaTrackingID && gaGtag` check, instead of having two of the same check?

Comment (JoelMarcey) [reply lib/core/Head.js]: Basically


```
        {this.props.config.gaTrackingId &&
          this.props.config.gaGtag && (
            <script
              async
              src={`https://www.googletagmanager.com/gtag/js?id=${
                this.props.config.gaTrackingId
              }`}
            />
            <script
              dangerouslySetInnerHTML={{
                __html: `
              window.dataLayer = window.dataLayer || [];
              function gtag(){dataLayer.push(arguments); }
              gtag('js', new Date());
              gtag('config', '${this.props.config.gaTrackingId}');
            `,
              }}
            />
          )}
```

Comment (JoelMarcey) [reply lib/core/Head.js]: Does that not work?

**Comment by j-f1** [comment]
Any idea when this will make it to a release?

**Comment by JoelMarcey** [comment]
@j-f1 I am going to try to get out a release this weekend.

**Pull request merged**